jekyll/_cci2/testing-macos.md

```diff
@@ -0,0 +1,222 @@
+
+While it can be written to manually with `sqlite3` commands, [CircleCI provides an Orb](https://circleci.com/orbs/registry/orb/circleci/macos) to simplify this.
+
+## Setting up a macOS UI Test Project
+
+Configuring CircleCI to run UI tests on a macOS app is split into two parts. Firstly, the CircleCI config needs to add the correct permissions and set up the environment to run the tests. Secondly, Fastlane needs to be configured to execute the tests.
+
+### Configuring CircleCI
+
+In the CircleCI `config.yml` we need to include the `circleci/macos` [orb](https://circleci.com/orbs/registry/orb/circleci/macos) and call the `macos/add-mac-uitest-permissions` step. This step ensures that the correct permissions are added to run Xcode UI tests on a macOS app.
+
+If additional permissions are required, you can find out how to set these up at the [macOS permission orb documentation](https://circleci.com/orbs/registry/orb/circleci/macos).
+
+Sample `config.yml` for testing a macOS app:
+
+```yaml
+version: 2.1
+
+orbs:
+    mac-permissions: circleci/macos
+
+jobs:
+  build-test:
+    macos:
+      xcode: 11.7.0
+    steps:
+        - checkout
+        - run: echo 'chruby ruby-2.7' >> ~/.bash_profile
+        - mac-permissions/add-mac-uitest-permissions
+        - run: bundle install
+        - run: bundle exec fastlane testandbuild
+
+workflows:
+    verify:
+        jobs:
+            - build-test
+```
+
+### Configuring Fastlane
+
+Fastlane allows us to avoid calling lengthy Xcode commands manually and instead write a simple configuration file to initiate the macOS app tests. With Fastlane we can build, sign (for testing) and test a macOS app. 
+
+A simple config can be found below. Note that this config relies on the project being configured as "Sign to Run Locally" and therefore we do not need to set up Fastlane Match. If your app requires signing to test, follow the [code signing documentation]({{ site.baseurl }}/2.0/ios-codesigning/) (while the code signing documentation talks about iOS, it is also applicable to macOS).
+
+```ruby
+# fastlane/Fastfile
+default_platform :mac
+
+platform :mac do
+  before_all do
+    setup_circle_ci
+  end
+
+  desc "Run tests"
+  lane :testandbuild do
+    scan
+  end
+end
+```
+
+A fully configured sample project can be found [on GitHub](https://github.com/CircleCI-Public/macos-orb).
+
+## Working with the macOS Orb
+
+The `TCC.db` file is simply an SQLite database, so this makes it easy to inject new permissions, or modify existing ones, during a job. 
+
+While it can be written to manually with `sqlite3` commands, we encourage the use of the [macOS orb](https://circleci.com/orbs/registry/orb/circleci/macos) simplify this. The examples in this section are all based on using the orb.
```

Comment:
```suggestion
While it can be written to manually with `sqlite3` commands, we encourage the use of the [macOS orb](https://circleci.com/orbs/registry/orb/circleci/macos) to simplify this. The examples in this section are all based on using the orb.
```